cmd/bucket-handlers.go

```diff
@@ -1174,6 +1194,99 @@ func (api objectAPIHandlers) PostPolicyBucketHandler(w http.ResponseWriter, r *h
 		}
 	}
 
+	if len(fanOutReq) > 0 {
+		// Fan-out requires no copying, and must be carried from original source
+		// https://en.wikipedia.org/wiki/Copy_protection so the incoming stream
+		// is always going to be in-memory as we cannot re-read from what we
+		// wrote to disk - since that amounts to "copying" from a "copy"
+		// instead of "copying" from source, we need the stream to be seekable
+		// to ensure that we can make fan-out calls concurrently.
+		buf := bytebufferpool.Get()
+		n, err := io.Copy(buf, pReader)
```

Comment:
```suggestion
		// Maximum allowed fan-out object size.
		const maxFanOutSize = 16 << 20

		buf := bytebufferpool.Get()
		n, err := io.Copy(buf, ioutil.HardLimitReader(pReader, maxFanOutSize))
```

Reply:
do we really want to limit here?

Reply:
@harshavardhana Yes, otherwise users can just fill up memory with data.